examples/compress_gzip.c

```diff
@@ -0,0 +1,134 @@
+#include <stdio.h>
+#include <sys/stat.h>
+#include <sys/time.h>
+#include <stdlib.h>
+#include <limits.h>
+#include <assert.h>
+#include <string.h>
+#include <stdint.h>
+#include <getopt.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <qatzip.h>
+
+static int compressFile(int in_file, int out_file)
+{
+    long input_file_size = 0;
+    long output_file_size = 0;
+    unsigned int src_buffer_size = 0;
+    unsigned int dst_buffer_size = 0;
+    unsigned char *src_buffer = NULL;
+    unsigned char *dst_buffer = NULL;
+    unsigned int bytes_read = 0;
+    int rc;
+    int is_compress = 1;
+    QzSession_T session = {0};
+    QzSessionParamsDeflate_T params;
+
+    rc = qzGetDefaultsDeflate(&params);
+    if (rc != QZ_OK) {
+        fprintf(stderr, "qzGetDefaultsDeflate failed\n");
+        return -1;
+    }
+
+    rc = qzInit(&session, 0);
+    if (rc != QZ_OK && rc != QZ_DUPLICATE) {
+        fprintf(stderr, "qzInit failed\n");
+        return -1;
+    }
+
+    // You can customize parameters(ie, level, pollingmode)
+    // before setup
+    params.common_params.polling_mode = QZ_BUSY_POLLING;
```

Comment:
Please setup the compression using 16KB chunk, L9 as Elliot requested in the email.

Reply:
Updated